platform/pensando/sonic-platform-modules-dpu/dpu/utils/dpu_db_util.py

```diff
@@ -0,0 +1,926 @@
+#!/usr/bin/env python3
+# {C} Copyright 2023 AMD Systems Inc. All rights reserved
+#############################################################################
+# Pensando
+#############################################################################
+
+import sys
+import signal
+import subprocess
+import threading
+import time
+from datetime import datetime
+import json
+import docker
+import redis
+import syslog
+from sonic_py_common import daemon_base, logger, syslogger
+from swsscommon.swsscommon import SonicV2Connector
+from sonic_py_common.task_base import ProcessTaskBase
+import grpc
+from google.protobuf.empty_pb2 import Empty
+
+SYSLOG_IDENTIFIER = 'dpu-db-utild'
+logger_instance = syslogger.SysLogger(SYSLOG_IDENTIFIER)
+
+def log_info(msg, also_print_to_console=False):
+    logger_instance.log_info(msg, also_print_to_console)
+
+def log_err(msg, also_print_to_console=False):
+    logger_instance.log_error(msg, also_print_to_console)
+
+try:
+    from health_checker.manager import HealthCheckerManager
+    from sonic_py_common import daemon_base
+    import sonic_platform
+    from sonic_platform.chassis import Chassis
+    from sonic_platform.helper import APIHelper
+    from pathlib import Path
+    grpc_files_path = (str(Path(sonic_platform.__file__).parent.absolute()))
+    if grpc_files_path not in sys.path:
+        sys.path.append(grpc_files_path)
+    from sonic_platform import oper_pb2, oper_pb2_grpc, system_pb2, system_pb2_grpc
+except Exception as e:
+    log_err(f'failed to load modules due to {e}')
+
+DPU_HEALTH_INFO_TABLE_NAME = 'DPU_STATE'
+SYSTEM_HEALTH_INFO_TABLE_NAME = 'SYSTEM_HEALTH_INFO'
+REBOOT_CAUSE_INFO_TABLE_NAME = 'REBOOT_CAUSE'
+VERSION_UPDATE_TABLE_NAME = 'PLATFORM|VERSION|CURRENT'
+TEMPERATURE_INFO_TABLE_NAME = 'TEMPERATURE_INFO'
+VOLTAGE_INFO_TABLE_NAME = 'VOLTAGE_INFO'
+CURRENT_INFO_TABLE_NAME = 'CURRENT_INFO'
+REDIS_CHASSIS_SERVER_PORT = 6380
+REDIS_CHASSIS_SERVER_IP = '169.254.200.254'
+CHASSIS_STATE_DB = 13
+LOCALHOST = "127.0.0.1"
+EVENT_PORT = 11360
+PDS_PORT = 11357
+NOT_AVAILABLE = 'N/A'
+ERR_UNKNOWN = 1
+SIGNALS_TO_NAMES_DICT = dict((getattr(signal, n), n) for n in dir(signal) if n.startswith('SIG') and '_' not in n)
+
+# operd event functions
+g_count = 0
+CRITICAL_EVENTS = [
+    2, 4, 6, 7, 11, 12, 15, 16, 17, 18, 19, 20, 4001, 4002
+]
+NETWORK_EVENTS = [
+    1001,   # "Port link is up"
+    1002    # "Port link is down"
+]
+
+# process handler
+
+def get_slot_id(chassis):
+        try:
+            if chassis == None:
+                chassis = Chassis()
+            slot = chassis.get_my_slot()
+        except Exception as e:
+            log_err("failed to fetch chassis slot id due to {e}")
+            slot = -1
+        slot_id = str(slot) if slot != -1 else 'UNDEFINED'
+        return slot_id
+
+exit_code = ERR_UNKNOWN
+
+#
+# SensorDBUpdater  ======================================================================
+#
+
+class SensorDBUpdater(logger.Logger):
```

Comment:
@shanshri can we use sensord or sensormon for updating voltage, current and temperature similar to other platforms?

Reply:
Yes we will remove it as this is not part of phase-I, later we will try to use sensormond.

Reply:
@shanshri seems still not removed

Reply:
Done